pkg/crc/machine/start.go

```diff
@@ -112,12 +112,57 @@ func (client *client) updateVMConfig(startConfig types.StartConfig, vm *virtualM
 	return nil
 }
 
-func growRootFileSystem(sshRunner *crcssh.Runner, preset crcPreset.Preset, persistentVolumeSize int) error {
-	rootPart, err := getrootPartition(sshRunner, preset)
+func restartHost(ctx context.Context, vm *virtualMachine, sshRunner *crcssh.Runner) error {
+	if err := vm.Driver.Stop(); err != nil {
+		return errors.Wrap(err, "Error re-starting machine at stop")
+	}
+	if err := startHost(ctx, vm); err != nil {
+		return errors.Wrap(err, "Error re-starting machine at start")
+	}
+	logging.Debug("Waiting until ssh is available")
+	if err := sshRunner.WaitForConnectivity(ctx, 300*time.Second); err != nil {
+		return errors.Wrap(err, "Failed to connect to the CRC VM with SSH -- virtual machine might be unreachable")
+	}
+	return nil
+}
+
+func ocpGetPVShiftSize(diskSize int, pvSize int) int {
+	defaultPvSize := constants.GetDefaultPersistentVolumeSize(crcPreset.OpenShift)
+	if pvSize > defaultPvSize {
+		return (diskSize - constants.DefaultDiskSize) - (pvSize - defaultPvSize)
+	}
+	return diskSize - constants.DefaultDiskSize
```

Comment:
> What we want to do is to align the last partition to the end of the disk, so we can do it without any knowledge of the default disk size, the disk size stored in json metadata

that is not the case always, sometime we want to move the partition not to the very end and leave space to increase the PV partition size

suppose `disk-size` is `40` and `persistent-volume-size` is `5` (all in GB) then after increasing the entire disk to 40GB from 31GB the last partition needs to be moved by 7GB (not 9GB) leaving 2GB of space at the end so the PV partition can grow to become 5GB (default size is 3GB)

Reply:
We could grow the partition before moving it since it's at the end, or we can take into account its new size desired size when moving it, this is fine as this is data we cannot know from the current layout.

My impression from the current is that we determine the offset we need to move the partition without ever looking at the actual on disk layout, but only using hardcoded constants and crc configuration values. If the disk layout (unexpectedly) goes out of sync with this, bad things will happen I think?

Reply:
i think we can replace `constant.DefaultDiskSize` and `defaultPvSize` with the current disk size and current pv partition size fetching it with `sfdisk`, but i think it'll be an edge case if the config values do not reflect the actual values

Reply:
added https://github.com/crc-org/crc/pull/4114/commits/d129b861633b41aadc73767b7b43959ccd0f94fb to address this 

i'll try to incorporate this into the original commit, but since the functions gets moved to another file, its taking a bit of time to do this